polaris/hub/client.py

```diff
@@ -162,17 +169,24 @@ def _base_request_to_hub(self, url: str, method: str, **kwargs):
             response.raise_for_status()
 
         except HTTPStatusError as error:
+            response_status_code = response.status_code
+
             # With an internal server error, we are not sure the custom error-handling code on the hub is reached.
-            if response.status_code == 500:
+            if response_status_code == 500:
                 raise
 
             # If not an internal server error, the hub should always return a JSON response
             # with additional information about the error.
             response = response.json()
             response = json.dumps(response, indent=2, sort_keys=True)
-            raise PolarisHubError(
-                f"The request to the Polaris Hub failed. See the error message below for more details:\n{response}"
-            ) from error
+
+            if response_status_code == 403:
+                raise PolarisCreateArtifactError() from error
+
+            if response_status_code == 404:
+                raise PolarisRetrieveArtifactError() from error
```

Comment:
**Suggestion:** As we've seen in this PR, I think the mapping between HTTP status codes and actions (i.e. create / retrieve) is not self-explanatory. I would add a comment briefly documenting this. We may need to revisit this in the future as the Hub and Client evolve (e.g. if we want to support updating artifacts through the client).